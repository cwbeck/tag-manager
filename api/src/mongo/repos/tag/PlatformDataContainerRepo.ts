import { injectable } from 'inversify';
import PlatformDataContainer from '../../models/tag/PlatformDataContainer';
import UnderPlatformRevisionControl from './abstractions/UnderPlatformRevisionControl';
import { IndexSpecification } from 'mongodb';

@injectable()
export default class PlatformDataContainerRepo extends UnderPlatformRevisionControl<PlatformDataContainer> {
    protected readonly auditEnabled = true;

    protected readonly indexes: IndexSpecification[] = [
        {
            background: false,
            key: {
                ___persisting_id: 1,
                _revision_id: 1,
            },
            unique: true,
        },
    ];
}
