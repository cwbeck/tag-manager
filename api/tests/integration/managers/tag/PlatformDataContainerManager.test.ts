import ManagerTestUtils from '../../../utils/ManagerTestUtils';
import TestUtils from '../../../utils/TestUtils';
import PlatformDataContainerRepo from '../../../../src/mongo/repos/tag/PlatformDataContainerRepo';
import Mock = jest.Mock;

describe('PlatformDataContainerManager', () => {
    beforeEach(ManagerTestUtils.beforeEachPrepareContainer);
    afterEach(TestUtils.afterEachRestoreContainerMocksAndDate);

    it('fetches a single platform', async () => {
        ManagerTestUtils.mockOrgUserWithViewAccess();
        ManagerTestUtils.mockOrgUserWithViewAccessOnPlatformRevision();

        const mockSutModel = TestUtils.buildMockPlatformDataContainer();
        const mockSutRepo = {} as PlatformDataContainerRepo;
        mockSutRepo.findByIdThrows = jest.fn().mockResolvedValue(mockSutModel);
        (ManagerTestUtils.mockRepoFromModelFactory as Mock).mockReturnValue(mockSutRepo);

        const server = TestUtils.createApolloServerForIntegration();

        const Q = `
            query {
                getPlatformDataContainer(id: "5edf4c48e9615d9a1c66bc2a") {
                    id
                    name
                }
            }
        `;

        // run query against the server and snapshot the output
        const res = await server.executeOperation({ query: Q });
        expect(res).toMatchSnapshot();
    });
});
