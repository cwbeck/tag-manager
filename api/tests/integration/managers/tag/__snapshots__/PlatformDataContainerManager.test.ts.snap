// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlatformDataContainerManager fetches a single platform 1`] = `
Object {
  "data": Object {
    "getPlatformDataContainer": Object {
      "id": "5edf4c48e9615d9a1c66cabb",
      "name": "test platform data container",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
